ibm/service/power/resource_ibm_pi_instance.go

```diff
@@ -772,8 +801,37 @@ func resourceIBMPIInstanceUpdate(ctx context.Context, d *schema.ResourceData, me
 			}
 		}
 	}
-	return resourceIBMPIInstanceRead(ctx, d, meta)
+	if d.HasChanges(Arg_IBMiCSS, Arg_IBMiPHA, Arg_IBMiRDSUsers) {
+		if d.Get("status") == "ACTIVE" {
+			log.Printf("the lpar is in the Active state, continuing with update")
+		} else {
+			_, err = isWaitForPIInstanceAvailable(ctx, client, instanceID, "OK")
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		sl := &models.SoftwareLicenses{}
+		sl.IbmiCSS = flex.PtrToBool(d.Get(Arg_IBMiCSS).(bool))
+		sl.IbmiPHA = flex.PtrToBool(d.Get(Arg_IBMiPHA).(bool))
+		ibmrdsUsers := d.Get(Arg_IBMiRDSUsers).(int)
+		if ibmrdsUsers < 0 {
+			return diag.Errorf("request with IBMi Rational Dev Studio property requires IBMi Rational Dev Studio number of users")
+		}
```

Comment:
Not a requirement but we should be using ValidateFunc
https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#validatefunc

Reply:
That's probablt a bit more than I want to change for this PR. Probably mark it as an enhancement.